### ISSUE TITLE
Porting of Provider Network Model

### ISSUE THREAD

**Issue by ChrisDeh** [nautobot/nautobot#724]
### Environment
* Python version: v3.7
* Nautobot version: v1.0.3

### Proposed Functionality

Implementation of Provider Networks as a Model to connect multiple circuits to as done in NetBox: https://github.com/netbox-community/netbox/issues/5986

### Use Case

Documentation of e.G. Transit Connections, Peering LANs, MPLS Networks with P2MP Connections, etc.

### Database Changes

* Adding a new table.
* extending circuit termination

### External Dependencies

None

**Comment by ChrisDeh** [comment]
Shall i submit a PR for implementation of this?

**Comment by glennmatthews** [comment]
Thanks for offering! If you're willing to do so, it would be quite welcome. :-)

**Comment by chadell** [comment]
We could also enrich a bit more the original Netbox Model, for instance adding a reference to IPAM Prefixes (optional) could help to model some Provider Networks.

**Comment by chadell** [comment]
@ChrisDeh have you started with this effort? Definitely, this is something we want to add to Nautobot, so we would like to push it forward.

@glennmatthews , taking as reference the current implementation in Netbox for [ProviderNetwork](https://github.com/netbox-community/netbox/blob/develop/netbox/circuits/models.py#L86), I would like to ask you if it would make sense to you if we add an extra `ManyToMany` relationship with `ipam.Prefix`?
Any other suggestion/consideration before starting?

**Comment by ChrisDeh** [comment]
@chadell I sadly not managed to find any time for starting with this. I've planned to go ahead with this in the second half of december.

**Comment by johannwagner** [comment]
I am a colleague of @ChrisDeh and will take care of this 👌🏻

**Comment by bryanculver** [comment]
Merged into `next`.
